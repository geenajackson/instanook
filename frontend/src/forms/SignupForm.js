import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";


function SignupForm({ registerUser }) {
    const INITIAL_STATE = { username: "", password: "", email: "", friendCode: "SW-" }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const [errors, setErrors] = useState(null);
    const history = useHistory();

    async function handleSubmit(evt) {
        evt.preventDefault();
        let res = await registerUser(formData);

        if (res) {
            setErrors(res.map(error => (
                <li key={error}>{error}</li>
            )));
            setFormData(INITIAL_STATE);
        }
        else history.push("/");
    };

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    return (
        <div className="SignupForm">
            <h1>Register Here!</h1>
            {errors ? <p>{errors}</p> : ""}
            <Form onSubmit={handleSubmit}>
                <FormGroup>
                    <Label htmlFor="username">Username: </Label>
                    <Input
                        id="username"
                        name="username"
                        type="text"
                        value={formData.username}
                        onChange={handleChange}
                    ></Input>
                    <Label htmlFor="password">Password: </Label>
                    <Input
                        id="password"
                        name="password"
                        type="password"
                        value={formData.password}
                        onChange={handleChange}
                    ></Input>
                    <Label htmlFor="email">Email: </Label>
                    <Input
                        id="email"
                        name="email"
                        type="text"
                        value={formData.email}
                        onChange={handleChange}
                    ></Input>
                    <Label htmlFor="friendCode">Friend Code: SW-XXXX-XXXX-XXXX</Label>
                    <Input
                        id="friendCode"
                        name="friendCode"
                        type="text"
                        value={formData.friendCode}
                        onChange={handleChange}
                    ></Input>
                    <Button type="submit">Register!</Button>
                </FormGroup>
            </Form>
        </div>
    )
}

export default SignupForm;