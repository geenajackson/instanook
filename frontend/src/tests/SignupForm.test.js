import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "../forms/SignupForm";

it("renders without crashing", function () {
    render(<SignupForm />);
});

it("matches snapshot with no jobs", function () {
    const { asFragment } = render(<SignupForm />);
    expect(asFragment()).toMatchSnapshot();
});