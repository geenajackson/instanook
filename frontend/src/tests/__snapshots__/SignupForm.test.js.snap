// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <h1>
      Register Here!
    </h1>
    <form
      class=""
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="username"
        >
          Username: 
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="password"
        >
          Password: 
        </label>
        <input
          class="form-control"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <label
          class="form-label"
          for="email"
        >
          Email: 
        </label>
        <input
          class="form-control"
          id="email"
          name="email"
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="friendCode"
        >
          Friend Code: SW-XXXX-XXXX-XXXX
        </label>
        <input
          class="form-control"
          id="friendCode"
          name="friendCode"
          type="text"
          value="SW-"
        />
        <button
          class="btn btn-secondary"
          type="submit"
        >
          Register!
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
